services/src/fl-root/floodlight-modules/src/main/java/org/openkilda/floodlight/command/flow/InstallIngressRuleCommand.java

```diff
@@ -0,0 +1,182 @@
+/* Copyright 2019 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.floodlight.command.flow;
+
+import static org.openkilda.messaging.Utils.ETH_TYPE;
+import static org.projectfloodlight.openflow.protocol.OFVersion.OF_15;
+
+import org.openkilda.floodlight.command.MessageWriter;
+import org.openkilda.floodlight.command.OfCommand;
+import org.openkilda.floodlight.command.meter.InstallMeterCommand;
+import org.openkilda.floodlight.error.SwitchOperationException;
+import org.openkilda.floodlight.error.UnsupportedSwitchOperationException;
+import org.openkilda.floodlight.service.FeatureDetectorService;
+import org.openkilda.messaging.MessageContext;
+import org.openkilda.messaging.model.Switch;
+import org.openkilda.messaging.model.Switch.Feature;
+import org.openkilda.model.OutputVlanType;
+import org.openkilda.model.SwitchId;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import lombok.Getter;
+import net.floodlightcontroller.core.IOFSwitch;
+import net.floodlightcontroller.core.module.FloodlightModuleContext;
+import org.projectfloodlight.openflow.protocol.OFFactory;
+import org.projectfloodlight.openflow.protocol.OFFlowAdd;
+import org.projectfloodlight.openflow.protocol.OFFlowMod;
+import org.projectfloodlight.openflow.protocol.OFFlowModFlags;
+import org.projectfloodlight.openflow.protocol.action.OFAction;
+import org.projectfloodlight.openflow.protocol.action.OFActions;
+import org.projectfloodlight.openflow.protocol.instruction.OFInstructionApplyActions;
+import org.projectfloodlight.openflow.protocol.instruction.OFInstructionMeter;
+import org.projectfloodlight.openflow.protocol.match.Match;
+import org.projectfloodlight.openflow.types.EthType;
+import org.projectfloodlight.openflow.types.OFPort;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Getter
+public class InstallIngressRuleCommand extends InstallTransitRuleCommand {
+
+    private final Long bandwidth;
+    private final Integer inputVlanId;
+    private final OutputVlanType outputVlanType;
+    private final Long meterId;
+
+    @JsonCreator
```

Comment:
This class is serializable to JSON, which means that it's supposed to be built and transferred by another modules than FL. Is it correct? If so then those modules get dependence on org.projectfloodlight.

Reply:
No, these "commands" are only for internal floodlight purposes. They are created by deserializing from "floodligh API" classes that are used by another modules